src/data/ieee-programming/index.ts

```diff
@@ -0,0 +1,272 @@
+    });
+
+    // Merge data of multiple sources for each year
+    years.forEach((y) => {
+      row[y] = sources.map((source) => row[source + ' ' + year]);
+    });
+  });
+
+
+  // basic data
+  let dataBuilder = builder(parsed.data)
+    .column(buildStringColumn('name'))
+    .column(buildCategoricalColumn('web'))
+    .column(buildCategoricalColumn('mobile'))
+    .column(buildCategoricalColumn('enterprise'))
+    .column(buildCategoricalColumn('embedded'))
+    .column(buildNumberColumn('NA 2018')); // NA column of 2018
+
+  // source matrices
+  sources.forEach((source) => {
+    dataBuilder = dataBuilder.column(buildNumberColumn(source, [0, NaN]).asArray(years.length));
+  });
+
+  // year matrices
+  years.forEach((year) => {
+    dataBuilder = dataBuilder.column(buildNumberColumn('' + year, [0, NaN]).asArray(sources.length));
+  });
+
+  dataBuilder = dataBuilder.ranking(
+  buildRanking()
+      .supportTypes()
+      .column('name')
+      .column('2018')
+      .sortBy('2018', 'desc')
+    );
+
+  dataBuilder = dataBuilder.ranking(
+    buildRanking()
+        .supportTypes()
+        .column('name')
+        .column('2017').column('2016').column('2015').column('2014')
+        .sortBy('2017', 'desc')
+      );
+
+  const lineup= dataBuilder.deriveColors().buildTaggle(${domVariable});
+  `;
+  },
+  build(node: HTMLElement) {
+    return import('raw-loader!./ieee_language_rankings.csv').then((content: any) => {
+      const csv: string = content.default ? content.default : content;
+      this.rawData = csv;
+      return parse(csv, {
+        dynamicTyping: true,
+        header: true,
+        skipEmptyLines: true
+      });
+    }).then((parsed: ParseResult) => {
+      const years = [2014, 2015, 2016, 2017, 2018]; // 5 years
+      const sources = ['Career Builder', 'Dice', 'Github (active)', 'Github (created)', 'Google Search', 'Google Trends', 'Hacker News', 'IEEE Xplore', 'Reddit (posts)', 'Stack Overflow (?s)', 'Stack Overflow (views)', 'Twitter']; // 12 sources
+
+      parsed.data.forEach((row) => {
+        // //Merge data of multiple years for each source
+        sources.forEach((source) => {
+          row[source] = years.map((y) => row[`${source} ${y}`]);
+        });
+
+        // Merge data of multiple sources for each year
+        years.forEach((y) => {
+          row[y] = sources.map((source) => row[`${source} ${y}`]);
+        });
+      });
+
+
+      // basic data
+      let dataBuilder = builder(parsed.data)
+        .column(buildStringColumn('name'))
+        .column(buildCategoricalColumn('web'))
+        .column(buildCategoricalColumn('mobile'))
+        .column(buildCategoricalColumn('enterprise'))
+        .column(buildCategoricalColumn('embedded'))
+        .column(buildNumberColumn('NA 2018')); // NA column of 2018
+
+      // source matrices
+      sources.forEach((source) => {
+        dataBuilder = dataBuilder.column(buildNumberColumn(source, [0, NaN]).asArray(years.length));
```

Comment:
instead of the length you can also provide the labels of the array such that you have meaningful labels instead of `Column 0`

Reply:
Great. Just added it.